File: 3241-divide-array-into-arrays-with-max-difference/divide-array-into-arrays-with-max-difference.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        vector<vector<int>> result;

        sort(nums.begin(), nums.end());

        for (int i=0; i<nums.size(); i+=3) {
            if (nums[i] + k < nums[i+2]) {
                return {};
            } else {
                result.push_back({nums[i], nums[i+1], nums[i+2]});
            }
        }

        return result;
    }
};

File: 3241-divide-array-into-arrays-with-max-difference/divide-array-into-arrays-with-max-difference.cs
public class Solution {
    public int[][] DivideArray(int[] nums, int k) {
        Array.Sort(nums);
        int[][] result = new int[nums.Count()/3][];
        int i = 0;
        for (int j=0; j<nums.Count(); j+=3) {
            if (nums[j]+k < nums[j+2]) {
                return new int[0][];
            } else {
                result[i] = new int[3];
                result[i][0] = nums[j];
                result[i][1] = nums[j+1];
                result[i][2] = nums[j+2];
                i++;
            }
        }

        return result;
    }
}